src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -272,7 +272,9 @@ public static boolean callSpawnEntitySpawner(final List<Entity> entities, final
             frame.addContext(EventContextKeys.SPAWN_TYPE, SpawnTypes.WORLD_SPAWNER);
 
             final SpawnEntityEvent event = SpongeEventFactory.createSpawnEntityEventSpawner(frame.getCurrentCause(), entities);
-            SpongeImpl.postEvent(event);
+            if (ShouldFire.SPAWN_ENTITY_EVENT_SPAWNER) {
```

Comment:
I'm almost certain that this is intrinsically checked by the tracker since things are only captured for spawning if the spawn events are being listened to.

Reply:
It is. 
These methods in the factory are only called if the event has listeners (unless they are doing some special handling).